### ISSUE TITLE
.lock files? Why and can it be turned off?

### ISSUE THREAD

**Issue by RicardoMusch** [cpenv/cpenv#73]
The introduction of the locking mechanism is causing major issues on machines where multipel users log in (or where rendernodes are running as a service).

Is there any more info on it?

**Comment by danbradham** [comment]
Oh weird, are they being sticky?

I introduced them to fix an issue on our farm when GPU rendering. To prevent multiple tasks on the same machine from attempting to localize at the same time. Making them opt-in would be appropriate if they're causing issues for you.

Is this on windows or linux?

**Comment by RicardoMusch** [comment]
Windows.

it seems that if the lock exists no extra process can even access the localized packages?
I.e. a user was logged onto a machine, localized packages, then didn't log out.
Other user logs on, can't open software because of the lock issue.

Same for machines that have a rendernode running in the bg.

**Comment by RicardoMusch** [comment]
Would it be easier solved by naming the package to something random or temporary while extracting/localizing and then rename the folder when done?

**Comment by danbradham** [comment]
Hmmm, I'll have to take a look. Just disabling the locks first is the quickest fix, then I can take a deeper look. I'll try to take care of it today / tomorrow. It needs to be applied in this repo, tk-cpenv, and autocpenv. Sorry for the inconvenience.

**Comment by danbradham** [comment]
Hey @RicardoMusch,

I've decided to make the lockfiles opt-in to make them minimally obtrusive, until I get a chance to debug them further.

The environment variable is CPENV_ENABLE_LOCKFILES and defaults to 0. Unfortunately I'm sort of slammed right now at work, so I did what I could quickly this evening. tk-cpenv and autocpenv have also been updated. Try them out and let me know if they fix your issue.

Best regards,
Dan
